src/planner/binder/statement/bind_create_table.cpp

```diff
@@ -104,6 +104,39 @@ static void BindConstraints(Binder &binder, BoundCreateTableInfo &info) {
 			    make_unique<BoundUniqueConstraint>(move(keys), move(key_set), unique.is_primary_key));
 			break;
 		}
+		case ConstraintType::FOREIGN_KEY: {
+			auto &fk = (ForeignKeyConstraint &)*cond;
+			D_ASSERT((fk.info.type == ForeignKeyType::FK_TYPE_FOREIGN_KEY_TABLE && !fk.info.pk_keys.empty()) ||
+			         (fk.info.type == ForeignKeyType::FK_TYPE_PRIMARY_KEY_TABLE && !fk.info.pk_keys.empty()) ||
+			         fk.info.type == ForeignKeyType::FK_TYPE_SELF_REFERENCE_TABLE);
+			if (fk.info.type == ForeignKeyType::FK_TYPE_SELF_REFERENCE_TABLE && fk.info.pk_keys.empty()) {
+				for (auto &keyname : fk.pk_columns) {
+					auto entry = info.name_map.find(keyname);
+					if (entry == info.name_map.end()) {
+						throw ParserException("column \"%s\" named in key does not exist", keyname);
+					}
+					fk.info.pk_keys.push_back(entry->second);
+				}
+			}
+			if (fk.info.fk_keys.empty()) {
+				for (auto &keyname : fk.fk_columns) {
+					auto entry = info.name_map.find(keyname);
+					if (entry == info.name_map.end()) {
+						throw ParserException("column \"%s\" named in key does not exist", keyname);
```

Comment:
This line does not appear to be covered in the code coverage report. Could you add a test that covers this line?